ktor-server/ktor-server-host-common/jvm/src/io/ktor/server/engine/EnvironmentUtilsJvm.kt

```diff
@@ -19,7 +19,8 @@ internal actual fun buildApplicationConfig(args: Array<String>): ApplicationConf
         .filterKeys { it.startsWith("-P:") }
         .mapKeys { it.key.removePrefix("-P:") }
 
-    val configFile = argumentsPairs["-config"]?.let { File(it) }
+    val profile = argumentsPairs["-profile"]
+    val configFile = argumentsPairs["-config"]?.let { File(it) } ?: profile?.let { File("application-${profile}.conf") }
```

Comment:
How to read profile configs from the local file system if the are stored not in the current working directory?

Reply:
Hello @Stexxe, what do you think about such a solution to a stated problem?
```
    val profile = argumentsPairs["-profile"]
    val configFilePath = argumentsPairs["-config"]?.let {
        profile?.let { profileName ->
            if (it.endsWith(".conf")) {
                return@let "${it.removeSuffix(".conf")}-${profileName}.conf";
            }
            it
        } ?: it
    }

    val configFile = configFilePath?.let { File(it) } ?: profile?.let { File("application-${it}.conf") }
```

Reply:
@Stexxe but isn't this case will be executed when user hasn't specified a custom path to config file, which means Ktor should load one from `resources` folder?

Reply:
@rsinukov according to the new logic it reads a config with a path provided by `-config` option or a profile config in the current working directory.

Reply:
@Moluram I don't like it because it relies on a config extension.

Reply:
@Stexxe exactly. So user either specifies full path, or profile. In case of profile, we should only read from the current directory. Does it make your original comment irrelevant or am I missing something?